[airthings] Add support for Airthings Wave Gen 1

dw-8 opened the pull request:
<!--
Thanks for contributing to the openHAB project!
Please describe the goal and effect of your PR here.
Pay attention to the below notes and to *the guidelines* for this repository.
Feel free to delete any comment sections in the template (starting with "<!--").
-->

<!-- TITLE -->

<!--
Please provide a PR summary in the *Title* above, according to the following schema:
- If related to one specific add-on: Mention the add-on shortname in square brackets
  e.g. "[exec]", "[netatmo]" or "[tesla]"
- If the PR is work in progress: Add "[WIP]"
- Give a short meaningful description in imperative mood
  e.g. "Add support for device XYZ" or "Fix wrongly handled exception"
  for a new add-on/binding: "Initial contribution"
Examples:
- "[homematic] Improve communication with weak signal devices"
- "[timemachine][WIP] Initial contribution"
- "Update contribution guidelines on new signing rules"
-->

<!-- DESCRIPTION -->

<!--
Please give a few sentences describing the overall goals of the pull request.
Give enough details to make the improvement and changes of the PR understandable
to both developers and tech-savy users.

Please keep the following in mind:
- What is the classification of the PR, e.g. Bugfix, Improvement, Novel Addition, ... ?
- Did you describe the PRs motivation and goal?
- Did you provide a link to any prior discussion, e.g. an issue or community forum thread?
- Did you describe new features for the end user?
- Did you describe any noteworthy changes in usage for the end user?
- Was the documentation updated accordingly, e.g. the add-on README?
- Does your contribution follow the coding guidelines:
  https://www.openhab.org/docs/developer/guidelines.html
- Did you check for any (relevant) issues from the static code analysis:
  https://www.openhab.org/docs/developer/bindings/#include-the-binding-in-the-build
- Did you sign-off your work:
  https://www.openhab.org/docs/developer/contributing.html#sign-your-work
-->

<!-- TESTING -->

<!--
Your Pull Request will automatically be built and available under the following folder:
https://openhab.jfrog.io/openhab/libs-pullrequest-local/org/openhab/

It is a good practice to add a URL to your built JAR in this Pull Request description,
so it is easier for the community to test your Add-on.
If your Pull Request contains a new binding, it will likely take some time
before it is reviewed and processed by maintainers.
That said, consider submitting your Add-on in the Eclipse IoT Marketplace
See this thread for more info:
https://community.openhab.org/t/24491

Don't forget to submit a thread about your Add-on in the openHAB community:
https://community.openhab.org/c/add-ons 
-->
Add support for Airthings Wave Gen 1

Signed-off-by: Davy Wong davy.wong.on+github@gmail.com

kaikreuzer commented on bundles/org.openhab.binding.bluetooth.airthings/src/main/resources/OH-INF/thing/airthings.xml:
```diff
@@ -69,6 +69,30 @@
+			<bridge-type-ref id="roaming" />
+			<bridge-type-ref id="bluegiga" />
+			<bridge-type-ref id="bluez" />
+		</supported-bridge-type-refs><label>Airthings Wave Gen 1</label><description>Smart Radon Monitor</description><channels>
```
Some line breaks seem to be missing here.
Please run a `mvn spotless:apply`.

dw-8 replied:
I made the corrections.

kaikreuzer commented on bundles/org.openhab.binding.bluetooth.airthings/src/main/java/org/openhab/binding/bluetooth/airthings/internal/AirthingsWaveGen1Handler.java:
```diff
@@ -0,0 +1,134 @@
+            switch (readSensor) {
+                case TEMPERATURE:
+                    dblResult = intFromBytes(rawdata[0], rawdata[1]) / 100D;
+                    logger.info("Parsed data 1: {}", String.format("[temperature=%.1f °C]", dblResult));
```
Please only use debug level for all those logs (here and below).

dw-8 replied:
I changed all logging to debug level only, as requested.

kaikreuzer reviewed (changes requested):
Thanks for this addition!
Just some very small review comments before we can merge.

dw-8 commented:
I messed up this PR.

I merged openhab:main into this PR and then reverted that commit, but these other commits are still showing as files being changed in this PR, as if I am reverting them all out of main.

Do I need to revert the revert to fix this?

I am new to using github, and getting myself all confused when and where to sync (fetch and merge) forks and pull requests.

dw-8 commented:
I reverted the revert, and it correctly showed only my files being changed, so then I issue the "git rebase HEAD~# --signoff" to clear the DCO signoff requirements, but, then all the other users' individual commits (instead of the merge commit) all came back into my PR again.

I reverted all those users' commits, and then issued another "git rebase HEAD~# --signoff" but it still shows extra files being changed.

What am I doing wrong?

dw-8 commented:
Well, making those revisions to two files became very messy with a lot of extra commits...
I accidentally merged a bunch of other commits into my PR, which I tried to revert and then revert them again.
I think I cleaned it all up, but now I have 3 extra reviewers because of those other commits.

dw-8 commented:
@kaikreuzer Are you okay with the changes I made in address your requested changes?

kaikreuzer reviewed (approved):
Yes, looks good to me, thank you!

kaikreuzer commented:
FTR, the formatting was still not ok, I have created https://github.com/openhab/openhab-addons/pull/11068 to fix it.

kaikreuzer commented:
FTR, the SAT tests were failing as well 😢 . Fixed it with https://github.com/openhab/openhab-addons/pull/11074.
